src/query.rs

```diff
@@ -252,28 +281,32 @@ mod tests {
                         )
                     };
 
-                assert!(
-                    run_query_on(&indexed_crate_new, &indexed_crate_new)
-                        .1
-                        .is_empty(),
-                    "{}",
-                    crate_pair
-                );
-                assert!(
-                    run_query_on(&indexed_crate_old, &indexed_crate_old)
-                        .1
-                        .is_empty(),
-                    "{}",
-                    crate_pair
-                );
+                let assert_no_false_positives_in_nonchanged_crate =
+                    |crate_: &VersionedIndexedCrate,
+                     crate_version: &str| {
+                         let output_pair = run_query_on(crate_, crate_);
+                         if !output_pair.1.is_empty() {
+                             let output_difference = pretty_format_output_difference(
+                                 query_name,
+                                 "Expected output (empty output)".to_string(),
+                                 BTreeMap::new(),
+                                 format!("Actual output ({}/{})", crate_pair, crate_version),
+                                 BTreeMap::from([output_pair]));
+                             panic!("Running a query on a crate that didn't change should always produce an empty output.\n{}\n", 
```

Comment:
This message states an invariant but doesn't mention how the invariant was broken, which is confusing. Did the crate change unexpectedly? Did the query fail to run and instead return an error? Or was the output non-empty?

Also, the invariant here can be stated more strongly: it's not just that the crate didn't change, but the rustdoc JSON input was identical between the baseline and current.

Reply:
Fixed in b58de954ac788d4a4eab30f13eba324cb670d4ee